src/csv.rs

```diff
@@ -0,0 +1,121 @@
+use std::io::Write;
+use std::{
+    fs::OpenOptions,
+    time::{Duration, SystemTime, UNIX_EPOCH},
+};
+
+use crate::{
+    cpu::CPU,
+    logger::{self, Interface, Logger},
+};
+
+pub struct CSVWriter {
+    log_size_cutoff: i32,
+    path: String,
+    enabled: bool,
+    logger: Logger,
+}
+
+trait Writer {
+    fn write(&mut self, column: Column);
+    fn init(&mut self, column: Column);
+}
+
+trait Writable {
+    fn to_csv(&mut self) -> String;
+}
+
+impl Writable for CPU {
+    fn to_csv(&mut self) -> String {
+        format!(
+            "{},{},{},{},{},{},{},{},{}\n",
+            SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap_or(Duration::new(0 as u64, 1 as u32))
+                .as_secs(),
+            self.name,
+            self.number,
+            self.max_freq,
+            self.min_freq,
+            self.cur_freq,
+            self.cur_temp,
+            self.cur_usage,
+            self.gov
+        )
+    }
+}
+
+pub struct Column {
+    cpus: Vec<CPU>,
```

Comment:
I think the csv writer would be best if it just writes the current CPUs to the file with append more instead of storing them indefinitely like the normal logger does. Try to have cpu logging require as little extra memory as possible.

Reply:
In this instance, a new column is generated for each log event. I made it a struct because we might want to add a new value to the output and we would want to keep that standardized. I don't think using this increases memory overhead because it's created and then immediately discarded just as passing CPUs into the function directly would do.

Reply:
Okay, that's cool then. Sounds good